### PR TITLE
Enhancing multi tenant sample

### PR DESCRIPTION
**Pull request by TiagoBrenck**
## Purpose
A multi-tenant app goes beyond setting `organizations` as tenant. There are more topics that are essential for a developer to know when developing a simple multi-tenant app.

With help @kalyankrishna1 's help, we enhanced the multi-tenant sample to cover the topics:

- /common endpoint
- Service principle provision on other tenants
- Data Partition
- Graph token issued by signed user's tenant

This new sample is not a level 100 anymore, since it brings more complex topics. In order to provide a better experience, it uses a SQL DB, but there is the option of running it using in memory DB (thanks to EntityFramework).

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

* Test the code
The DB will be created automatically. If you want to run it using `InMemory` DB, read the line 48 on `Startup.cs`

Ideally you will want to have 2 tenants to test it.

## What to Check
Verify that the following are valid

- You can onboard different tenants and sign-in users (common endpoint and SP provisioning)
- You can CRUD todo items but cannot see items from other tenants (Data partition)
- When editing your todo item, a list of active users from your tenant is shown (Graph AT by tenant)

### CODE REVIEW

**Comment by jennyf19** [comment]
> using Microsoft.AspNetCore;

do you need license info for samples?

---
Refers to: 2-WebApp-graph-user/2-3-Multi-Tenant/Program.cs:1 in ca3aa52. [](commit_id = ca3aa52d07c0e376904362506e817e86df0fc228, deletion_comment = False)

File: 2-WebApp-graph-user/2-3-Multi-Tenant/README.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+# Build a multi-tenant SaaS web application that calls Microsoft Graph using Azure AD & OpenID Connect
+
+> This sample is for Azure AD, not Azure AD B2C.
+
+[![Build status](https://identitydivision.visualstudio.com/IDDP/_apis/build/status/AAD%20Samples/.NET%20client%20samples/ASP.NET%20Core%20Web%20App%20tutorial)](https://identitydivision.visualstudio.com/IDDP/_build/latest?definitionId=819)
+
+## About this sample
+
+This sample shows how to build an ASP.NET Core MVC web application that usesOpenID Connect to sign in users from multiple Azure AD tenants. Additionally it also introduces developers to the concept of a [multi-tenant](https://docs.microsoft.com/azure/active-directory/develop/single-and-multi-tenant-apps) Azure Active Directory application.
+
```
Comment (jmprieur) [inline 2-WebApp-graph-user/2-3-Multi-Tenant/README.md]: Maybe we want to mention why it calls the graph?

File: 1-WebApp-OIDC/1-2-AnyOrg/AppCreationScripts/sample.json [hunk 2]
```diff
@@ -20,6 +20,13 @@
       "HomePage": "https://localhost:44321/",
       "ReplyUrls": "https://localhost:44321/, https://localhost:44321/signin-oidc",
       "LogoutUrl": "https://localhost:44321/signout-oidc",
+      "PasswordCredentials": "Auto",
+      "RequiredResourcesAccess": [
+        {
+          "Resource": "Microsoft Graph",
```
[the remark this answers is not in the document]
Comment (TiagoBrenck) [reply 1-WebApp-OIDC/1-2-AnyOrg/AppCreationScripts/sample.json]: @jmprieur are you sure you have the latest version? I reverted this change already.

**Comment by TiagoBrenck** [comment]
@jmprieur I am using the Graph SKD. It is on [MSGraphService.cs](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/blob/tibre/multiTenantSample/2-WebApp-graph-user/2-3-Multi-Tenant/BLL/MSGraphService.cs)

**Comment by jmprieur** [comment]
my bad, @TiagoBrenck

**Comment by TiagoBrenck** [comment]
Ok @jmprieur I will change. Me and Kalyan also discussed a better flow for the onboarding on the navigation menu, so I will update it as well

**Pull request merged**